src/gt4py/frontend/gtscript_frontend.py

```diff
@@ -669,12 +705,6 @@ def visit_If(self, node: ast.If):
 #
 
 
-@enum.unique
-class ParsingContext(enum.Enum):
-    CONTROL_FLOW = 1
-    COMPUTATION = 2
-
-
 class IRMaker(ast.NodeVisitor):
```

Comment:
With the addition of the parallel intervals, it's very hard to understand how `with` statements are parsed. Creating an additional cannonicalization from python ast -> python ast that transforms into a flat representation would make sense. However since the frontend is to be refactored anyway at some point I'm fine with this right now.